### PR TITLE
NIAD-2791

### PR DESCRIPTION
**Pull request by oleksiiRybak**
FHIR Immunization Extention adjustment which is required based on the contract examples.
https://developer.nhs.uk/apis/gpconnect-1-6-0/accessrecord_structured_development_fhir_examples_immunizations.html

### CODE REVIEW

File: gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapperTest.java [hunk 1]
```diff
@@ -88,21 +87,35 @@ public void mapObservationToImmunizationWhenEhrCompositionIdIsNotPresentOnEncoun
...
     }
 
+    @Test
+    public void mapObservationToImmunizationWithMultipleObservationStatementsAndCodingDisplay() {
+        final String VACCINE_PROCEDURE_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-VaccinationProcedure-1";
+        var ehrExtract = unmarshallEhrExtract("full_valid_immunization_with_multiple_observation_statements.xml");
+        List<Immunization> immunizationList = immunizationMapper.mapResources(ehrExtract, getPatient(), getEncounterList(), PRACTISE_CODE);
+
+        var immunization = (Immunization) immunizationList.get(0);
+        assertThat(immunizationList).hasSize(THREE);
```
Comment (adrianclay) [inline gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapperTest.java:106]: I don't know about anybody else in the team, but I'm a fan of the Single Assert rule.

In other words, this test should only be checking for one behaviour.

At the moment, we've got several asserts and from the test name I'm not clear what this test should be specifically checking for.

Comment (oleksiiRybak) [reply gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapperTest.java]: yes, agree that the test is cluttered with other not-relevant test asserts hence I adjusted it.

File: gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java [hunk 2]
```diff
@@ -38,8 +38,7 @@
 @AllArgsConstructor
 public class ImmunizationMapper extends AbstractMapper<Immunization> {
     private static final String META_PROFILE = "Immunization-1";
-    private static final String VACCINE_PROCEDURE_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect"
-        + "-VaccinationProcedure-1";
+    private static final String VACCINE_PROCEDURE_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-VaccinationProcedure-1";
     private static final String END_DATE_PREFIX = "End Date: ";
     private static final String VACCINATION_CODING_EXTENSION_URL = "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-coding"
```
Comment (benhession) [inline gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java:43]: Is this constant being used now or can it be removed?

Comment (oleksiiRybak) [reply gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java]: not used and is removed now, thanks, good catch.

**Pull request merged**